propka/output.py

```diff
@@ -1,397 +1,491 @@
-      str += "Could not determine folding profile\n"
+            if group.residue_type == residue_type:
+                str_ += "{0:s}".format(
+                    group.get_summary_string(
+                        parameters.remove_penalised_group))
+    return str_
+
+
+def get_folding_profile_section(protein, conformation='AVR',
+                                direction="folding", reference="neutral",
+                                window=[0., 14., 1.0], _=False,
+                                __=None):
+    """Returns string with the folding profile section of the results.
+
+    Args:
+        protein:  protein object
+        conformation:  specific conformation
+        direction:  'folding' or other
+        reference:  reference state
+        window:  pH window [min, max, step]
+        _:  Boolean for verbose output
+        __:  options object
+    Returns:
+        string
+    """
+    str_ = get_the_line()
+    str_ += "\n"
+    str_ += "Free energy of {0:>9s} (kcal/mol) as a function".format(direction)
+    str_ += " of pH (using {0:s} reference)\n".format(reference)
+    profile, [ph_opt, dg_opt], [dg_min, dg_max], [ph_min, ph_max] = (
+        protein.get_folding_profile(
+            conformation=conformation, reference=reference,
+            grid=[0., 14., 0.1]))
+    if profile is None:
+        str_ += "Could not determine folding profile\n"
     else:
-      for (pH, dG) in profile:
-        if pH >= window[0] and pH <= window[1] and (pH%window[2] < 0.05 or pH%window[2] > 0.95):
-          str += "%6.2lf%10.2lf\n" % (pH, dG)
-      str += "\n"
-
-    if pH_opt == None or dG_opt == None:
-      str += "Could not determine pH optimum\n"
+        for (ph, dg) in profile:
+            if ph >= window[0] and ph <= window[1]:
+                if ph % window[2] < 0.05 or ph % window[2] > 0.95:
+                    str_ += "{0:>6.2f}{1:>10.2f}\n".format(ph, dg)
+        str_ += "\n"
+    if ph_opt is None or dg_opt is None:
+        str_ += "Could not determine pH optimum\n"
     else:
-      str += "The pH of optimum stability is %4.1lf for which the free energy is%6.1lf kcal/mol at 298K\n" % (pH_opt, dG_opt)
-
-    if dG_min == None or dG_max == None:
-      str += "Could not determine pH values where the free energy is within 80 %s of minimum\n" % ("%")
+        str_ += "The pH of optimum stability is {0:>4.1f}".format(ph_opt)
+        str_ += (
+            " for which the free energy is {0:>6.1f} kcal/mol at 298K\n".format(
+                dg_opt))
+    if dg_min is None or dg_max is None:
+        str_ += "Could not determine pH values where the free energy"
+        str_ += " is within 80 %% of minimum\n"
     else:
-      str += "The free energy is within 80 %s of maximum at pH %4.1lf to %4.1lf\n" % ("%", dG_min, dG_max)
-
-    if pH_min == None or pH_max == None:
-      str += "Could not determine the pH-range where the free energy is negative\n\n"
+        str_ += "The free energy is within 80 %% of maximum"
```

Comment:
should probably by a single `%`

Reply:
Fixed in https://github.com/Electrostatics/propka-3.1/commit/83a7099d79999b49e880fdf854010c8f0b0ab2fd